code/scripting/lua/LuaReference.h

```diff
@@ -101,7 +101,7 @@ class UniqueLuaReference {
     * @brief Pushes the referenced value onto the stack.
     * @param thread A specific thread state to push the value to. nullptr for the default state of this reference
     */
-	void pushValue(lua_State* thread) const;
+	void pushValue(lua_State* thread, bool guaranteeStack = false) const;
```

Comment:
While I make this call in this instance, it is necessary to do this _for every case_ this function is called. Which is from quite a lot of places. You have to compromise at some point, either by asking the coder to always manually ensure the stack size before each call of pushValue or pushReference, or eat the performance penalty of doing the allocation / check more than once if you push many values, or have the option to pass this as the API parameter.
Since I would really not want to have to rely on the programmer to always remember manually ensuring the stack size, it's basically a question of "do we want to waste performance in a low level function" vs "do we want one more parameter in the low level function".
IMO, with this case we should pick performance over pretty API.

Reply:
ok, clarified variable name and added documentation